opensrs/opensrsapi.py

```diff
@@ -237,16 +237,90 @@ def _send_authcode(self, domain_name):
         return self._req(action='SEND_AUTHCODE', object='DOMAIN',
                          attributes={'domain_name': domain_name})
 
-    def _renew_domain(self, domain_name, current_expiration_year, period):
+    def _register_domain(self, domain, purchase_period, user, user_id,
+                         password, nameservers=None, private_reg=False,
+                         reg_domain=None, extras=None,
+                         order_processing_method=OrderProcessingMethods.SAVE):
+        extras = extras or {}
+        attrs = self._make_domain_reg_attrs(
+            domain, purchase_period, user, user_id, password, nameservers,
+            private_reg, reg_domain,
+            order_processing_method=order_processing_method, **extras)
+        if extras:
+            attrs.update(extras)
+
+        try:
+            rsp = self._sw_register_domain(attrs)
+        except errors.XCPError, e:
+            if e.response_code == self.CODE_DOMAIN_REGISTRATION_TAKEN:
+                raise errors.DomainTaken(e)
+            if e.response_code == self.CODE_DOMAIN_REGISTRATION_FAILED:
+                if (e.response_text.startswith('Invalid domain syntax') or
+                        e.response_text.startswith(
+                            'Invalid syntax on domain')):
+                    raise errors.InvalidDomain(e)
+                raise errors.DomainRegistrationFailure(e)
+            if e.response_code == self.CODE_CLIENT_TIMED_OUT:
+                raise errors.DomainRegistrationTimedOut(e)
+            raise
```

Comment:
This try...except logic around registrar call looks a bit ugly.  What do you think about moving it to some decorator?